Source/Tools/FEXLoader/LinuxSyscalls/GdbServer.cpp

```diff
@@ -1289,46 +1293,45 @@ void GdbServer::StartThread() {
 }
 
 void GdbServer::OpenListenSocket() {
-  // getaddrinfo allocates memory that can't be removed.
-  FEXCore::Allocator::YesIKnowImNotSupposedToUseTheGlibcAllocator glibc;
-  struct addrinfo hints, *res;
-
-  memset(&hints, 0, sizeof(hints));
-  hints.ai_family = AF_UNSPEC;
-  hints.ai_socktype = SOCK_STREAM;
-  hints.ai_flags = AI_PASSIVE;
-
-  if(getaddrinfo(NULL, "8086", &hints, &res) < 0) {
-    perror("getaddrinfo");
+  const auto GdbUnixPath = fextl::fmt::format("{}/FEX_gdbserver/", FEXServerClient::GetServerMountFolder());
+  if (FHU::Filesystem::CreateDirectory(GdbUnixPath) == FHU::Filesystem::CreateDirectoryResult::ERROR) {
```

Comment:
This will fail if any of the parent directories doesn't exist. Would `CreateDirectories` be more appropriate to use here?

Reply:
It wouldn't be more appropriate. The folder that this ends up in is either `/tmp/` or `/run/user/<uid>/` and we don't want to try and create any of those parent directories.

If the user used the XDG directories to override the path, then they also need to make sure those folders exist otherwise regular applications break.